### PR TITLE
[docs] Tweak std.variant docs

### PR DESCRIPTION
**Pull request by ntrel**
See commit messages.

### CODE REVIEW

**Comment by ntrel** [comment]
Tests fail for unrelated reason AFAICT:

    timelimit: sending warning signal 15
    posix.mak:358: recipe for target 'unittest/std/regex/internal/tests.run' failed

File: std/variant.d [hunk 1]
```diff
@@ -2092,18 +2093,19 @@ unittest
  *
  * Returns: The return type of tryVisit is deduced from the visiting functions and must be
  * the same across all overloads.
- * Throws: If no parameter-less, error function is specified: $(D VariantException) if
- *         $(D variant) doesn't hold a value or
+ * Throws: $(D VariantException) if no parameter-less fallback function is specified.
+ *         Thrown when $(D variant) doesn't hold a value or
```
Comment (Hackerpilot) [inline std/variant.d]: Suggestion:
```
* Throws: $(D VariantException if no parameter-less fallback function is specified,
*         $(D variant) doesn't hold a value, or 
```

File: std/variant.d [hunk 2]
```diff
@@ -1991,13 +1991,14 @@ unittest
  * Throws: If no parameter-less, error function is specified:
  * $(D VariantException) if $(D variant) doesn't hold a value.
  */
-template visit(Handler ...)
-    if (Handler.length > 0)
+template visit(Handlers ...)
```
Comment (Hackerpilot) [inline std/variant.d]: :white_check_mark:

File: std/variant.d [hunk 3]
```diff
@@ -1969,19 +1969,19 @@ unittest
...
  * in the template parameter list.
- * It is statically ensured that all types of
+ * It is statically ensured that all held types of
  * $(D variant) are handled across all handlers.
  * $(D visit) allows delegates and static functions to be passed
  * as parameters.
  *
  * If a function without parameters is specified, this function is called
- * when variant doesn't hold a value. Exactly one parameter-less function
+ * when `variant` doesn't hold a value. Exactly one parameter-less function
```
Comment (Hackerpilot) [inline std/variant.d]: Maybe "when the variant doesn't hold a value"?

**Comment by JackStouffer** [comment]
Ping @ntrel

File: std/variant.d (hunk 1 above)
Comment (ntrel) [inline std/variant.d]: I've reworded this in 04ab0a0, should be clear now.

File: std/variant.d (hunk 3 above)
Comment (ntrel) [inline std/variant.d]: `variant` (in quotes) refers to the runtime argument of `visit` - the existing docs use this style, e.g. on line 1979 - only here they forgot to add the `$(D )` quotes.

**Comment by JackStouffer** [comment]
the uses of VariantException should be LREFed as well IMO

**Comment by ntrel** [comment]
> uses of VariantException should be LREFed as well

Done. I also added some other useful LREFs and small edits to the general `std.variant` docs in a separate commit.

File: std/variant.d [hunk 4]
```diff
@@ -1424,22 +1423,10 @@ unittest
     assert(b[1] == 3.14);
 }
 
-/** Code that needs functionality similar to the $(D boxArray)
-function in the $(D std.boxer) module can achieve it like this:
-*/
-unittest
```
Comment (JackStouffer) [inline std/variant.d]: Why was this removed?

Comment (ntrel) [inline std/variant.d]: `std.boxer` doesn't (seem to) exist, for some time. The example code also doesn't actually provide anything specific over what variadic template functions provide.

**Comment by Hackerpilot** [comment]
Auto-merge toggled on
